src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/PEImage.cs

```diff
@@ -76,6 +79,10 @@ private PEImage(Stream stream, bool leaveOpen, bool isVirtual)
             if (!stream.CanSeek)
                 throw new ArgumentException($"{nameof(stream)} is not seekable.");
 
+            _memoryStream = new MemoryStream();
```

Comment:
If the original stream is seekable, we could probably pre-allocate with stream.Length.